sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/CallToAsyncMethodShouldNotBeBlocking.cs

```diff
@@ -122,6 +125,17 @@ private static void ReportOnViolation(SyntaxNodeAnalysisContext context)
                 messageArgs: MemberNameToMessageArguments[memberAccessNameName]));
         }
 
+        private static bool IsResultInContinueWithCall(string memberAccessName, SyntaxNode syntaxNode)
+        {
+            if (memberAccessName != ResultName)
+            {
+                return false;
+            }
+
+            var parentInvocation = syntaxNode.FirstAncestorOrSelf<InvocationExpressionSyntax>();
```

Comment:
I'm afraid that this will stop on first wrapping Invocation and cause FNs like this
``` C#
return continuation().ContinueWith((action) =>
{
    var t = action.Result.Run(null, token); // This is now OK
    return t.Result; // This should still be noncompliant
});
```
or FPs when there would be wrapper like `CheckResult(action.Result)` inside the call

Reply:
That is a very strange case when a task has an action as a result.

Reply:
My concern is about scenario, when `ContinueWith` calls any other async operation, retrieving it's task and calling that secondary `task.Result` => violating the rule.

Reply:
The implementation will not work for wrapper inside call:
```C#
return task.ContinueWith(completedTask =>
{
    return string.Format("Result: {0}", completedTask.Result);
});
```

Reply:
Better finding of `ContinueWith`, extracting it's argument name and validating it against the `variableName `that` .Result` was invoked on should do the trick. Or checking symbol type to be IParameterSymbol (or how is it called) with small performance penalty for symbol extraction

Reply:
I understand now what you mean and the example makes sense. I've updated the code to look up until it finds a ContinueWith method with the right argument name. Shadowing will give FN but I think that's ok for now.